rackspace/metrics/v2/query/requests.go

```diff
@@ -0,0 +1,140 @@
+package query
+
+import (
+	"github.com/mitchellh/mapstructure"
+	"github.com/rackspace/gophercloud"
+)
+
+const (
+	SUM      = "sum"
+	MAX      = "max"
+	MIN      = "min"
+	LATEST   = "latest"
+	VARIANCE = "variance"
+	AVERAGE  = "averge"
+)
+
+//Roll-ups process full-resolution data into coarser granularities of 5 minutes, 20 minutes, 60 minutes, 4 hours and 24 hours.
+const (
+	FULL    = "full"
+	MIN5    = "min5"
+	MIN20   = "min20"
+	MIN60   = "min60"
+	MIN240  = "min240"
+	MIN1440 = "min1440"
+)
+
+type QueryParams struct {
+	From       int64    `q:"from"`
+	To         int64    `q:"to"`
+	Until      int64    `q:"until"`
+	Points     int32    `q:"points"`
+	Resolution string   `q:"resolution"`
+	Select     []string `q:"select"`
+	Query      string   `q:"query"`
+	Tags       string   `q:"tags"`
+}
+
+// GetDataForListByPoints retrieve data against a list of metrics and number of points, for the specified tenant associated with RackspaceMetrics.
+func GetDataForListByPoints(c *gophercloud.ServiceClient, opts QueryParams, metrics ...string) (MetricListData, error) {
```

Comment:
All functions should return some kind of `Result` and then extract the content via an `Extract` method in `results.go`.